In [7]:
import numpy as np
import pandas as pd

In [ ]:
taxons = pd.read_csv('../../data/clean_taxons.csv')

In [8]:
content = pd.read_csv('../../data/clean_content.csv')

In [9]:
taxons = taxons[['base_path','content_id','taxon_name','level1taxon','level2taxon','level3taxon','level4taxon']]

In [10]:
content_taxons = pd.merge(
    left=content, 
    right=taxons, 
    left_on='taxon_id', 
    right_on='content_id', 
    how='outer', 
    indicator=True
)

In [11]:
content_taxons.head()

Unnamed: 0                          base_path_x  \
0         4.0                       /bank-holidays   
1        10.0  /calculate-your-holiday-entitlement   
2        78.0      /maternity-paternity-calculator   
3        95.0        /calculate-statutory-sick-pay   
4    201278.0     /training-study-work-your-rights   

                           content_id_x  \
0  58f79dbd-e57f-4ab2-ae96-96df5767d1b2   
1  deedf6f8-389b-4b34-a5b1-faa9ef909a70   
2  05d5412d-455b-485e-a570-020c9176a46e   
3  1c676a9e-0424-4ebb-bab8-d8cb8d2fc6f8   
4  f50efa51-dc62-4111-aa59-731af77806aa   

                                         description  \
0                                                      
1  holiday calculator to work out statutory holid...   
2  calculate an employee’s maternity pay (smp) pa...   
3  statutory sick pay (ssp) calculator - calculat...   
4  your rights to request time off for training o...   

                                             details document_type  \
0                                                 {}      calendar   
1  {'transaction_start_link': '/calculate-your-ho...   transaction   
2  {'transaction_start_link': '/maternity-paterni...   transaction   
3  {'transaction_start_link': '/calculate-statuto...   transaction   
4  {'external_related_links': [], 'parts': [{'slu...         guide   

              first_published_at locale primary_publishing_organisation  \
0  2016-02-29T09:24:10.000+00:00     en                             NaN   
1  2016-02-29T09:24:10.000+00:00     en                             NaN   
2  2016-02-29T09:24:10.000+00:00     en                             NaN   
3  2016-02-29T09:24:10.000+00:00     en                             NaN   
4  2016-02-29T09:24:10.000+00:00     en                             NaN   

  publishing_app  ...   variable                              taxon_id  \
0      calendars  ...        0.0  ffeb8bc6-2cd3-4a73-ab76-072b9a1f95fb   
1   smartanswers  ...        0.0  ffeb8bc6-2cd3-4a73-ab76-072b9a1f95fb   
2   smartanswers  ...        0.0  ffeb8bc6-2cd3-4a73-ab76-072b9a1f95fb   
3   smartanswers  ...        0.0  ffeb8bc6-2cd3-4a73-ab76-072b9a1f95fb   
4      publisher  ...        0.0  ffeb8bc6-2cd3-4a73-ab76-072b9a1f95fb   

          base_path_y                          content_id_y  \
0  /business/time-off  ffeb8bc6-2cd3-4a73-ab76-072b9a1f95fb   
1  /business/time-off  ffeb8bc6-2cd3-4a73-ab76-072b9a1f95fb   
2  /business/time-off  ffeb8bc6-2cd3-4a73-ab76-072b9a1f95fb   
3  /business/time-off  ffeb8bc6-2cd3-4a73-ab76-072b9a1f95fb   
4  /business/time-off  ffeb8bc6-2cd3-4a73-ab76-072b9a1f95fb   

                     taxon_name level1taxon       level2taxon level3taxon  \
0  Statutory leave and time off    Business  Employing people         NaN   
1  Statutory leave and time off    Business  Employing people         NaN   
2  Statutory leave and time off    Business  Employing people         NaN   
3  Statutory leave and time off    Business  Employing people         NaN   
4  Statutory leave and time off    Business  Employing people         NaN   

  level4taxon _merge  
0         NaN   both  
1         NaN   both  
2         NaN   both  
3         NaN   both  
4         NaN   both  

[5 rows x 23 columns]

In [12]:
content_taxons.shape

(340457, 23)

In [14]:
content.shape

(336950, 15)

In [16]:
taxons.shape

(5517, 7)

In [15]:
content_taxons['_merge'].value_counts()

both          335931
left_only       2915
right_only      1611
Name: _merge, dtype: int64

There are 2,915 content items tagged to taxons which are not present in the taxons data.  
There are 1,611 taxons which have not been tagged to in any of the 336,950 content items.  

This needs to be explored and is likely to be due to taxons being exported from 'draft content store' and content being exported from 'rummager', which may not be in sync. Check with devs

## Check out duplicates and missing

In [24]:
content_taxons[content_taxons.duplicated(keep=False)]

Unnamed: 0                                        base_path_x  \
6788          64.0  /government/news/proposed-new-timetable-for-st...   
6789          64.0  /government/news/proposed-new-timetable-for-st...   
6790        1434.0  /government/publications/state-pension-age-rev...   
6791        1434.0  /government/publications/state-pension-age-rev...   
6792        1627.0  /government/publications/state-pension-age-tim...   
6793        1627.0  /government/publications/state-pension-age-tim...   
6794        2508.0  /government/news/new-timetable-for-state-pensi...   
6795        2508.0  /government/news/new-timetable-for-state-pensi...   
6796        3443.0   /government/collections/state-pension-age-review   
6797        3443.0   /government/collections/state-pension-age-review   
6798        5512.0  /government/publications/state-pension-age-ind...   
6799        5512.0  /government/publications/state-pension-age-ind...   
6800        8268.0  /government/news/john-cridland-cbe-appointed-t...   
6801        8268.0  /government/news/john-cridland-cbe-appointed-t...   
6802       21180.0  /government/news/john-cridland-cbe-and-the-gov...   
6803       21180.0  /government/news/john-cridland-cbe-and-the-gov...   
6804       24094.0  /government/publications/state-pension-age-per...   
6805       24094.0  /government/publications/state-pension-age-per...   
6806       31127.0  /government/publications/2010-to-2015-governme...   
6807       31127.0  /government/publications/2010-to-2015-governme...   
6808       35819.0  /government/news/john-cridland-cbe-launches-co...   
6809       35819.0  /government/news/john-cridland-cbe-launches-co...   
6810       38294.0  /government/consultations/when-the-state-pensi...   
6811       38294.0  /government/consultations/when-the-state-pensi...   
6812       38651.0  /government/consultations/state-pension-age-in...   
6813       38651.0  /government/consultations/state-pension-age-in...   
6814       41376.0  /government/publications/state-pension-age-rev...   
6815       41376.0  /government/publications/state-pension-age-rev...   
6816       60813.0  /government/publications/long-term-state-pensi...   
6817       60813.0  /government/publications/long-term-state-pensi...   
...            ...                                                ...   
333064    613578.0                  /report-a-lost-or-stolen-passport   
333065    613578.0                  /report-a-lost-or-stolen-passport   
336320    814971.0                          /child-benefit-move-to-uk   
336321    814971.0                          /child-benefit-move-to-uk   
336322    815067.0       /child-benefit-child-lives-with-someone-else   
336323    815067.0       /child-benefit-child-lives-with-someone-else   
336324    816102.0                              /child-benefit-abroad   
336325    816102.0                              /child-benefit-abroad   
336326    816220.0                               /child-benefit-16-19   
336327    816220.0                               /child-benefit-16-19   
336328    817432.0    /child-benefit-for-children-in-hospital-or-care   
336329    817432.0    /child-benefit-for-children-in-hospital-or-care   
336330    817506.0                      /report-changes-child-benefit   
336331    817506.0                      /report-changes-child-benefit   
336349    815973.0                          /child-benefit-tax-charge   
336350    815973.0                          /child-benefit-tax-charge   
336351    817295.0                      /child-benefit-tax-calculator   
336352    817295.0                      /child-benefit-tax-calculator   
336364    816192.0                              /child-benefit-number   
336365    816192.0                              /child-benefit-number   
336366   1021683.0           /claim-child-benefit-behalf-someone-else   
336367   1021683.0           /claim-child-benefit-behalf-someone-else   
336368   1430527.0                               /guardians-allowa

In [ ]:
content_taxons[content_taxons.duplicated()].shape